core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala

```diff
@@ -174,18 +196,44 @@ trait ContainerUtils extends Logging {
     private def runDockerCmd(skipLogError: Boolean, args: Seq[String])(implicit transid: TransactionId): DockerOutput =
         ContainerUtils.runDockerCmd(dockerhost, skipLogError, args)(transid)
 
-    // If running outside a container, then logs files are in docker's own
-    // /var/lib/docker/containers.  If running inside a container, is mounted at /containers.
-    // Root access is needed when running outside the container.
-    private def dockerContainerDir(mounted: Boolean) = {
-        if (mounted) "/containers" else "/var/lib/docker/containers"
+    /**
+     * Obtain the per container directory Docker maintains for each container.
+     *
+     * If this component is running outside a container, then logs files are in docker's own
+     * /var/lib/docker/containers.  If running inside a container, is mounted at /containers.
+     * Root access is needed when running outside the container.
+     */
+    private def dockerContainerDir(mounted: Boolean, containerId: ContainerHash) = {
+        (if (mounted) "/containers/" else "/var/lib/docker/containers/") + containerId.hash.toString
     }
 
     /**
      * Gets the filename of the docker logs of other containers that is mapped back into the invoker.
      */
     private def getDockerLogFile(containerId: ContainerHash, mounted: Boolean) = {
-        new java.io.File(s"""${dockerContainerDir(mounted)}/${containerId.hash}/${containerId.hash}-json.log""").getCanonicalFile()
+        new java.io.File(s"""${dockerContainerDir(mounted, containerId)}/${containerId.hash}-json.log""").getCanonicalFile()
+    }
+
+    /**
+     * Return docker config as a JsObject by reading the config.v2.json file for the given container.
+     */
+    private def getDockerConfig(containerId: ContainerHash, mounted: Boolean): JsObject = {
+        val configFile = s"${dockerContainerDir(mounted, containerId)}/config.v2.json"
+        val contents = scala.io.Source.fromFile(configFile).mkString
+        contents.parseJson.asJsObject
+    }
+
+    /**
+     * Extracts the IP addr from the docker config object with projection rather than converting the entire config.
+     */
+    private def getIpAddr(config: JsObject): Option[String] = {
+        val networks = config.fields("NetworkSettings").asJsObject.fields("Networks").asJsObject
+        val userland = networks.fields("userland").asJsObject
+        val ipAddr = userland.fields("IPAddress")
+        ipAddr match {
+            case JsString(s) => Some(s)
+            case _ => None
+        }
```

Comment:
As you're ignoring exception throwing above, you might aswell do

`val ipAddr = userland.fields("IPAdress").convertTo[String]` instead of the matching